File: cpp/OOP/humanClass.py
class human:
    def __init__(self, name="unknown", age=0):
        print("Human object is created ")
        self.name = name
        self.age = age
    def sayName(self):
        print("Name is : %s" % self.name)
    def sayAge(self):
        print("Age is : %i" % self.age)
    def __del__(self):
        print("human object is deleted ")

File: cpp/OOP/humanClass.cpp
// the humanClass implementation cpp file
// we put the implementation here
#include"humanClass.h" // in the same folder

human::human( string name, int age){
    cout <<"Constructor initiated" << endl;
    this -> name = name;
    this -> age = age;
}
void human::sayName(){
    cout << "Name is : " << name <<endl;
}

void human::sayAge(){
    cout << "Age is : " << age << endl;
}
human::~human(){
    cout << "Human destrector exec"<<endl;
}